src/tables/sortable-table.js

```diff
@@ -36,37 +36,118 @@ import classnames from 'classnames'
-  controlled: PropTypes.bool.isRequired,
-  data: PropTypes.arrayOf(PropTypes.object).isRequired,
-  disableSort: PropTypes.bool.isRequired,
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string,
+  data: PropTypes.arrayOf(PropTypes.object),
+  initialAscending: PropTypes.bool,
+  initialColumn: PropTypes.string,
+  disableReverse: PropTypes.bool,
+  disableSort: PropTypes.bool,
+  controlled: PropTypes.bool,
+  onChange: PropTypes.func,
   rowComponent: Types.component,
   headerComponent: Types.component,
-  className: PropTypes.string,
-  ...sortablePropTypes,
 }
-
 const defaultProps = {
-  className: ''
+  className: '',
+  data: [],
+  initialAscending: true,
+  initialColumn: '',
+  disableReverse: false,
+  disableSort: false,
+  controlled: false,
+  onChange: noop,
 }
 
-function SortableTable ({
-  columns,
+function getInitialSortControls(initialColumn, columns) {
+  const defaultControls = {
+    initialSortPath: '',
+    initialSortFunc: null,
+    initialValueGetter: null,
+  }
+  if (!initialColumn) return defaultControls
+
+  const initialProps = columns.filter(col => col.name === initialColumn).pop()
+  // Exceptional situation-- an initial column was specified but no column data
+  // exists for the named column...
+  if (!initialProps) throw new TableColumnError('initial column has no column definition')
+
+  return {
+    initialSortPath: initialProps.name,
+    initialSortFunc: initialProps.sortFunc,
+    initialValueGetter: initialProps.valueGetter,
+  }
+}
+
+
+function SortableTable({
+  className,
+  children,
   data: unsortedData,
+  initialAscending,
+  initialColumn,
+  disableReverse,
   disableSort,
   controlled,
-  sort,
-  ascending,
-  sortPath,
-  setSortPath,
-  setSortFunc,
+  onChange,
   rowComponent,
   headerComponent,
-  className,
 }) {
-  const data = (controlled || disableSort) ? unsortedData : sort(unsortedData)
+  const columns = getColumnData(children, disableSort)
+  const { initialSortPath, initialSortFunc, initialValueGetter } =
+    getInitialSortControls(initialColumn, columns)
+  const [ascending, setAscending] = useState(initialAscending)
+  const [sortPath, setSortPath] = useState(initialSortPath)
+
+  // Setting and storing a function object requires special syntax.
+  // See: https://medium.com/swlh/how-to-store-a-function-with-the-usestate-hook-in-react-8a88dd4eede1
+  const [sortFunc, setSortFunc] = useState(() => initialSortFunc)
+  const [valueGetter, setValueGetter] = useState(() => initialValueGetter)
+
+  const data = useMemo(() => {
+    if (controlled || disableSort) return unsortedData
+
+    if (sortFunc) {
+      const sorted = [...unsortedData].sort(sortFunc)
+      if (!ascending && !disableReverse) sorted.reverse()
+      return sorted
+    }
+    else {
+      const order = ascending ? 'asc' : 'desc'
+      const sorted = orderBy(
+        unsortedData,
+        (item) => valueGetter ? valueGetter(item) : get(sortPath, item),
```

Comment:
Since this now uses `valueGetter`, you should add that to the dependencies array of `useMemo`.

Reply:
Good one. Fixed.